### PR TITLE
Use provider entities on recordings microservice

### PR DESCRIPTION
**Pull request by danigargar**
Replaced use of TrunksCrd with BillableCalls

#### Type Of Change 
- [x] Small bug fix
- [ ] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [x] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [x] Commits are split per component (schema, portal/platform, kamusers, agis, ..)
- [x] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

### CODE REVIEW

**Review by rbatistadev — APPROVED** [review]

File: library/spec/Ivoz/Provider/Domain/Service/BillableCall/UpdateDtoByDefaultRunTpCdrSpec.php [hunk 1]
```diff
@@ -146,6 +146,10 @@ private function prepareExecution()
             BillableCallDto::class
         );
 
+//        $this->billableCallDto
+//            ->getStartTime()
+//            ->willReturn(new \DateTime('2000-01-01 00:00:00'));
+
```
Comment (Kaian) [inline library/spec/Ivoz/Provider/Domain/Service/BillableCall/UpdateDtoByDefaultRunTpCdrSpec.php:152]: Commented code is not allowed

File: microservices/recordings/src/Service/Encoder.php [hunk 2]
```diff
@@ -3,13 +3,12 @@
 namespace Service;
 
 use Ivoz\Core\Domain\Service\EntityTools;
-use Ivoz\Kam\Domain\Model\TrunksCdr\TrunksCdrInterface;
-use Ivoz\Kam\Domain\Model\TrunksCdr\TrunksCdrRepository;
 use Ivoz\Kam\Domain\Model\UsersCdr\UsersCdrInterface;
```
Comment (Kaian) [inline microservices/recordings/src/Service/Encoder.php:6]: Wrong namespace. We want the interface with same name in Ivoz\Provider\

**Review by Kaian — CHANGES_REQUESTED** [review]

File: library/spec/Ivoz/Provider/Domain/Service/BillableCall/UpdateDtoByDefaultRunTpCdrSpec.php (hunk 1 above)
Comment (danigargar) [reply library/spec/Ivoz/Provider/Domain/Service/BillableCall/UpdateDtoByDefaultRunTpCdrSpec.php]: This code belongs to the parent branch where it is already fixed.
I have rebased both branches and this commit is gone.

File: microservices/recordings/src/Service/Encoder.php [hunk 3]
```diff
@@ -174,7 +173,9 @@ public function processAction(): void
                     }
                 } elseif ($kamAccCdr instanceof UsersCdrInterface) {
                     $type = RecordingInterface::TYPE_ONDEMAND;
-                    if ($kamAccCdr->getXcallid()) {
+
+                    $direction = $kamAccCdr->getDirection();
+                    if ($direction === 'outbound') {
```
Comment (Kaian) [inline microservices/recordings/src/Service/Encoder.php:178]: you can use UsersCdrInterface::DIRECTION_OUTBOUND here

**Review by Kaian — CHANGES_REQUESTED** [review]

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/UsersCdrDoctrineRepository.php [hunk 4]
```diff
@@ -37,4 +37,17 @@ public function findByKamUsersCdrId(int $id): ?UsersCdrInterface
             'kamUsersCdr' => $id,
         ]);
     }
+
+    /**
+     * @return UsersCdr[]
+     */
+    public function findByCallid(string $callid)
```
Comment (rbatistadev) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/UsersCdrDoctrineRepository.php:44]:
```suggestion
    public function findByCallid(string $callid): array
```

**Review by ironArt3mis — APPROVED** [review]

**Pull request merged**